src/muc_light/mod_muc_light_room.erl

```diff
@@ -185,8 +185,8 @@ process_config_set(#config{ raw_config = [{<<"subject">>, _}] } = ConfigReq, Roo
 process_config_set(_ConfigReq, _RoomUS, member, _AffUsers, false) ->
     {error, not_allowed};
 process_config_set(ConfigReq, {_, RoomS} = RoomUS, _UserAff, AffUsers, _AllCanConfigure) ->
-    case mod_muc_light_room_config:apply_binary_kv(
-           ConfigReq#config.raw_config, [], mod_muc_light:config_schema(RoomS)) of
+    case mod_muc_light_room_config:from_binary_kv_diff(
```

Comment:
Why do we need this new function here? Because we don't want to store the default values? I'm not sure if I understand this change, as there are places which used `apply_binary_kv` and run `from_binary_kv` now, not `apply_binary_kv_diff`.

Reply:
There are exactly two types of conversions from binary KV:
- Conversion of the initial config. Here all missing values should be substituted with defaults.
- Conversion of the config change (diff). Here all missing values should be skipped.

In the past there was a function called `apply_binary_kv` which was used for both cases and merged two configs. The point is that we never want to merge any two configs, so for each case I checked which of the two conversions needs to happen and explicitly called that one, making the code easier to understand and to optimize. moreover, there is now no need for storing the default config anymore.

Reply:
After the discussion with @gustawlippa I have to add that I omitted the third type of conversions, which is used by the RDBMS backend when reading the data from the DB. I used `from_binary_kv` there because it is more natural, but there should be no missing values in the DB, so no defaults will be injected anyway.